CodeU_Assignement_1/src/codeu_assignement_1/SinglyLinkedList.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package codeu_assignement_1;
+
+import java.util.Collection;
+import java.util.Iterator;
+
+/**
+ *
+ * @author Karine
+ * 
+ * Bounds, index and general interface is as in Java LinkedList JavaDoc
+ * https://docs.oracle.com/javase/7/docs/api/java/util/LinkedList.html
+ * 
+ */
+public class SinglyLinkedList <E> {
+    /*  Inner pojo class Node in the list  */
+    public class Node <T>
+    {
+        protected T data;
+        protected Node next;
+
+        /*  Default C'tor  */
+        public Node()
+        {
+            next = null;
+            data = null;
+        }    
+        /*  C'tor  */
+        public Node(T d,Node n)
+        {
+            data = d;
+            next = n;
+        }    
+        // Setters
+        public void setNext(Node n) { next = n; }    

```

Comment:
I prefer to have getters and setters and keep it as classic POJO; but maybe it is unique to Java/C not sure how it is in other programming languages.